portal-api/src/v1/users/sign-in-link.controller.js

```diff
@@ -1,10 +1,45 @@
+const { LOGIN_STATUSES } = require('../../constants');
+const utils = require('../../crypto/utils');
+const { updateLastLogin } = require('./controller');
+const { sanitizeUser } = require('./sanitizeUserData');
+
 class SignInLinkController {
   #signInLinkService;
 
   constructor(signInLinkService) {
     this.#signInLinkService = signInLinkService;
   }
 
+  async validateSignInLink(req, res) {
+    try {
+      const { signInToken } = req.params;
+      const { user } = req;
+
+      const isValidSignInToken = await this.#signInLinkService.isValidSignInToken({ username: user.username, signInToken });
+
+      if (!isValidSignInToken) {
+        throw new Error(`Invalid sign in token for user: ${user.username}`);
+      }
```

Comment:
I think we want this case to cause a 403 error instead of a 500, what do you think?